app/controllers/trips_controller.rb

```diff
@@ -0,0 +1,28 @@
+class TripsController < ApplicationController
+skip_before_action :authenticate_user!, only: [ :create, :show ]
+before_action :set_trip, only: [ :show ]
+
+  def create
+    @trip = Trip.new(trip_params)
+    # assign user_id to trips if user is logged in
+    @trip.user = current_user
+    if @trip.save
+      redirect_to trip_path(@trip)
+    else
+      render 'pages/home'
+    end
+  end
+
+  def show
```

Comment:
Don't create unnecessary methods. Keep it simple.

Reply:
Hey Kleyv. Exactly. Everyone should be able to create a trip and see the country page. But when bookmark or notification a visitor has to login first to become a user